pkg/operator/profilecalculator.go

```diff
@@ -519,42 +516,10 @@ func (pc *ProfileCalculator) tunedsUsePodLabels(tunedSlice []*tunedv1.Tuned) boo
 	return false
 }
 
-// This can be simplified and cleaned up a lot after
-// https://github.com/openshift/hypershift/pull/1702 merges.
-// Then the nodes will have a label with the NodePool name.
+// getNodePoolNameForNode returns the NodePool name from a label on the hosted cluster Node
 func (pc *ProfileCalculator) getNodePoolNameForNode(node *corev1.Node) (string, error) {
-	var nodeMachineSetName string
-	var resourceGVK schema.GroupVersionResource
-
-	if node.Annotations[hypershiftNodeOwnerKindLabel] == "MachineSet" {
-		nodeMachineSetName = node.Annotations[hypershiftNodeOwnerNameLabel]
-		resourceGVK = schema.GroupVersionResource{
-			Group:    "cluster.x-k8s.io",
-			Version:  "v1beta1",
-			Resource: "machinesets"}
-	} else {
-		klog.Warning("unexpected node owner kind on Node %s: %s", node.Name, hypershiftNodeOwnerKindLabel)
-	}
-
-	unstructuredMachineSet, err := pc.clients.ManagementDynamic.Resource(resourceGVK).Namespace(config.OperatorNamespace()).Get(context.TODO(), nodeMachineSetName, metav1.GetOptions{})
-	if err != nil {
-		return "", fmt.Errorf("error getting MachineSet for Node %s: %v", node.Name, err)
-	}
-
-	machineSetMetadata, ok := unstructuredMachineSet.Object["metadata"].(map[string]interface{})
-	if !ok {
-		return "", fmt.Errorf("unable to get annotations of unstructured MachineSet for node: %s", node.Name)
-	}
-	machineSetAnnotations, ok := machineSetMetadata["annotations"].(map[string]interface{})
-	if !ok {
-		return "", fmt.Errorf("unable to get metadata of unstructured MachineSet for node: %s", node.Name)
-	}
-	nodePoolNamespacedName, ok := machineSetAnnotations[hypershiftNodePoolNamespacedNameLabel].(string)
-	if !ok {
-		return "", fmt.Errorf("unable to get nodePool annotation value of unstructured MachineSet for node: %s", node.Name)
-	}
-	nodePoolName := parseNamespacedName((nodePoolNamespacedName))
-	klog.Infof("calculated nodePoolName: %s for node %s with nodeMachineSet value: %s", nodePoolName, node.Name, nodeMachineSetName)
+	nodePoolName := node.GetLabels()[hypershiftNodePoolLabel]
+	klog.Infof("calculated nodePoolName: %s for node %s", nodePoolName, node.Name)
```

Comment:
Can we align (perhaps in the future PR) the verbosity with other loggin in `profilecalculator.go`?  This verbosity is too high -- I missed this in the previous review.